test/fixtures/sqlite/lock-db.js

```diff
@@ -0,0 +1,12 @@
+const { DatabaseSync } = require('node:sqlite');
+
+const dbPath = process.argv[2];
+const conn = new DatabaseSync(dbPath);
+
+conn.exec('BEGIN EXCLUSIVE TRANSACTION');
+process.send('locked');
+setTimeout(() => {
```

Comment:
I think we should try to avoid relying on `setTimeout()` as that will almost certainly make the tests prone to being flaky or inefficient in the CI. I think we could rely on message passing (or hopefully move this logic out of a child process and into the same process as the test) instead of a timeout.